modules/communications/communications_worker.py

```diff
@@ -0,0 +1,66 @@
+"""
+Logs data and forwards it.
+"""
+
+import os
+import pathlib
+import queue
+
+from . import communications
+from utilities.workers import queue_proxy_wrapper
+from utilities.workers import worker_controller
+from ..common.modules.logger import logger
+
+
+def communications_worker(
+    timeout: float,
+    home_position_queue: queue_proxy_wrapper.QueueProxyWrapper,
+    input_queue: queue_proxy_wrapper.QueueProxyWrapper,
+    output_queue: queue_proxy_wrapper.QueueProxyWrapper,
+    controller: worker_controller.WorkerController,
+) -> None:
+    """
+    Worker process.
+
+    home_position_queue contains home positions for creating communications object.
+    input_queue and output_queue are data queues.
+    controller is how the main process communicates to this worker process.
+    """
+
+    worker_name = pathlib.Path(__file__).stem
+    process_id = os.getpid()
+    result, local_logger = logger.Logger.create(f"{worker_name}_{process_id}", True)
+    if not result:
+        print("ERROR: Worker failed to create logger")
+        return
+
+    # Get Pylance to stop complaining
+    assert local_logger is not None
+
+    local_logger.info("Logger initialized", True)
+
+    # Get home position
+    try:
+        home_position = home_position_queue.queue.get(timeout=timeout)
+    except queue.Empty:
```

Comment:
Are you sure that this is the exception when it times out? I'm not too sure, @Xierumeng

Reply:
Yes, this is correct.